test/ng/directive/ngOptionsSpec.js

```diff
@@ -922,6 +922,25 @@ describe('ngOptions', function() {
       expect(scope.selected).toEqual([20]);
       expect(element).toEqualSelectValue([20], true);
     });
+
+
+    it('should ignore array values beginning with $$', function() {
+      scope.arr = ['one', 'two', '$$hashKey'];
```

Comment:
The problem is that this can happen when your ngOptions values are also repeated over by ngRepeat - see this plnkr: http://plnkr.co/edit/eXyWQp9jH2AyflxSCTqL?p=preview This definitely worked differently in 1.3.x